File: ios/MapManager.m
#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MapManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onMapPress, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(showUserLocationPoint, BOOL)
RCT_EXPORT_VIEW_PROPERTY(customRegion, NSDictionary)

RCT_EXTERN_METHOD(searchLocation:(nonnull NSNumber *)node text:(NSString *)text resolve:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(searchCoodinate:(nonnull NSNumber *)node query:(NSString *)query resolve:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(annotate:(nonnull NSNumber *)node coodinate:(NSDictionary *)coodinate)
RCT_EXTERN_METHOD(removeAllAnnotations:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(showCircle:(nonnull NSNumber *)node config:(NSDictionary *)config)
RCT_EXTERN_METHOD(removeCurrentCircle:(nonnull NSNumber *)node)
@end

File: ios/MapManager.swift
import Foundation
import UIKit

@objc(MapManager)
class MapManager: RCTViewManager {
  let mapView = MapView()
  
  override func view() -> UIView! {
    return mapView
  }
  
  @objc
  func searchLocation(_ text: String!, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    mapView.searchLocation(text, resolve: resolve, rejecter: reject)
  }
  
  @objc
  func searchCoodinate(_ query: String!, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    mapView.searchCoodinate(query, resolve: resolve, rejecter: reject)
  }
  
  @objc
  func annotate(_ coodinate: [String: Double]) {
    mapView.annotate(coodinate)
  }
  
  @objc
  func removeAllAnnotations() {
    mapView.removeAllAnnotations()
  }
  
  @objc
  func showCircle(_ config: [String: Double]) {
    mapView.showCircle(config)
  }
  
  @objc
  func removeCurrentCircle() {
    mapView.removeCurrentCircle()
  }
  
  func getMapView(tag: NSNumber) -> MapView {
    return bridge.uiManager.view(forReactTag: tag) as! MapView
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
